corehq/apps/app_manager/static/app_manager/js/app_manager.js

```diff
@@ -518,23 +532,136 @@ hqDefine('app_manager/js/app_manager', [
+                        var existingCaseTypes = data.existing_case_types;
+                        self.deprecatedCaseTypes(data.deprecated_case_types);
+
+                        // Add existing case types as options
+                        existingCaseTypes.forEach(function (caseType) {
+                            var option = new Option(caseType, caseType, false, false);
+                            $caseType.append(option);
+                        });
+                    },
+                });
+
+                $caseType.on('change', function () {
+                    var valueNoSpaces = $(this).val();
+                    var $formGroup = $(this).closest('.form-group');
+                    var $help = $('#new_case_type_help');
+                    var $error = $('#new_case_type_error');
+                    var $createBtn = $('#case-type-create-btn');
+
+                    // Reset error states
+                    $formGroup.removeClass('has-error');
+                    $help.show();
+                    $error.hide();
+                    $createBtn.prop('disabled', false);
+
+                    if (!valueNoSpaces) {
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+
+                    if (module.valueIncludeInvalidCharacters(valueNoSpaces)) {
+                        $formGroup.addClass('has-error');
+                        $error.text(module.invalidCharErrorMsg);
+                        $error.show();
+                        $help.hide();
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+
+                    if (module.valueIsReservedWord(valueNoSpaces)) {
+                        $formGroup.addClass('has-error');
+                        $error.text(module.reservedWordErrorMsg);
+                        $error.show();
+                        $help.hide();
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+
+                    if (self.deprecatedCaseTypes().includes(valueNoSpaces)) {
+                        $formGroup.addClass('has-error');
+                        $error.text(module.deprecatedCaseTypeErrorMsg);
+                        $error.show();
+                        $help.hide();
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+                });
             }
+        });
+
+        // Handle "Create Application" button click
+        $('#case-type-create-btn').on('click', function () {
+            google.track.event("Added Case List Menu");
+            kissmetrix.track.event("Added Case List Menu");
+
+            var $caseTypeInput = $('#new_case_type');
+            var value = $caseTypeInput.val();
+
+            var $form = $('#new-module-form');
+            $('#new-case-type').val(value);
```

Comment:
Seeing `new_case_type` vs `new-case-type` is a bit confusing. Is there some way we can clarify this?

Reply:
If possible, we may also want to try to standardize `_` vs `-` (if we are manually creating ids with both). If these are HTML attributes we are defining ourselves, it seems our preference is to use `-`

Reply:
fd57d576880b3083a995d1f84be12fa73029cbf9